backoff/backoff.go

```diff
@@ -0,0 +1,70 @@
+package backoff
+
+import (
+	"math/rand"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	DefaultInterval    = time.Second
+	DefaultMaxInterval = time.Minute
+)
+
+// ExponentialBackoff provides jittered exponential durations for the purpose of
+// avoiding flodding a service with requests.
+type ExponentialBackoff struct {
+	Interval time.Duration
+	Max      time.Duration
+
+	currentInterval atomic.Value
+	cancel          <-chan struct{}
+}
```

Comment:
Mixing exported and non-exported fields _can_ be OK, but then you also mix direct and constructor initialization for this type, and combined I think that's a recipe for trouble...